api/src/identity-access-management/domain/usecases/anonymize-gar-authentication-methods.usecase.js

```diff
@@ -0,0 +1,15 @@
+/**
+ * @typedef {function} updateGARAuhenticationMethods
+ * @param {Object} params
+ * @param {string} params.userIds
+ * @param {AuthenticationMethodRepository} params.authenticationMethodRepository
+ * @return {Promise<void>}
+ */
+export const anonymizeGARAuthenticationMethods = async function ({ userIds, authenticationMethodRepository }) {
+  const total = userIds.length;
+  const { linesUpdated: anonymizedUserIds } = await authenticationMethodRepository.batchAnonymizeByUserIds({
+    userIds: userIds,
+  });
+
+  return { anonymized: anonymizedUserIds, total };
```

Comment:
Le nommage ci-dessous me parait plus explicite : 

```suggestion
  const { linesUpdated: anonymizedUserCount } = await authenticationMethodRepository.batchAnonymizeByUserIds({
    userIds: userIds,
  });

  return { anonymizedUserCount, total };
```